### PR TITLE
Add reuse test make target

### PR DESCRIPTION
**Pull request by jharrington22**
The PR adds a `test-reuse` make target to test AWS account reuse.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -136,4 +164,25 @@ delete-ccs-accountclaim:
...
+	@export AWS_ACCESS_KEY_ID=$(shell oc get secret osd-creds-mgmt-osd-staging-1-osdmanagedadminsre-secret -n aws-account-operator -o json | jq -r '.data.aws_access_key_id' | base64 -d); \
+	export AWS_SECRET_ACCESS_KEY=$(shell oc get secret osd-creds-mgmt-osd-staging-1-osdmanagedadminsre-secret -n aws-account-operator -o json | jq -r '.data.aws_secret_access_key' | base64 -d); \
+	aws s3api create-bucket --bucket ${REUSE_BUCKET_NAME} --region=us-east-1
+
+# List S3 bucket
+.PHONY: list-s3-bucket
+list-s3-bucket:
+	# Get credentials
+	export AWS_ACCESS_KEY_ID=$(shell oc get secret osd-creds-mgmt-osd-staging-1-osdmanagedadminsre-secret -n aws-account-operator -o json | jq -r '.data.aws_access_key_id' | base64 -d); \
+	export AWS_SECRET_ACCESS_KEY=$(shell oc get secret osd-creds-mgmt-osd-staging-1-osdmanagedadminsre-secret -n aws-account-operator -o json | jq -r '.data.aws_secret_access_key' | base64 -d); \
```
Comment (clcollins) [inline Makefile:182]: Just a suggestion for getting the secrets without using `jq` or `base64 -d`:

`oc extract secret/osd-creds-mgmt-osd-staging-1-osdmanagedadminsre-secret -n aws-account-operator --keys=aws_secret_access_key --to=- 2>&1 | tail -n 1 `

Just throwing it out there as an option.

File: Makefile [hunk 2]
```diff
@@ -54,6 +58,30 @@ delete-account: delete-account-secrets
...
+# Delete account claim namespace
+.PHONY: delete-account-claim-namespace
+delete-accountclaim-namespace:
+	# Delete namespace
+	@oc process -p NAME=${ACCOUNT_CLAIM_NAMESPACE} -f hack/templates/namespace.tmpl | oc delete -f -
+
+.PHONY: create-accountclaim
+create-accountclaim:
+	# Create accountclaim
+	@oc process -p NAME=${ACCOUNT_CLAIM_NAME} -p NAMESPACE=${ACCOUNT_CLAIM_NAMESPACE} -f hack/templates/aws_v1alpha1_accountclaim_cr.tmpl | oc apply -f -
```
[the remark this answers is not in the document]
Comment (jharrington22) [reply Makefile:76]: Fixed

File: Makefile [hunk 3]
```diff
@@ -136,4 +164,25 @@ delete-ccs-accountclaim:
...
+	# Get credentials
+	@export AWS_ACCESS_KEY_ID=$(shell oc get secret osd-creds-mgmt-osd-staging-1-osdmanagedadminsre-secret -n aws-account-operator -o json | jq -r '.data.aws_access_key_id' | base64 -d); \
+	export AWS_SECRET_ACCESS_KEY=$(shell oc get secret osd-creds-mgmt-osd-staging-1-osdmanagedadminsre-secret -n aws-account-operator -o json | jq -r '.data.aws_secret_access_key' | base64 -d); \
+	aws s3api create-bucket --bucket ${REUSE_BUCKET_NAME} --region=us-east-1
+
+# List S3 bucket
+.PHONY: list-s3-bucket
+list-s3-bucket:
+	# Get credentials
+	export AWS_ACCESS_KEY_ID=$(shell oc get secret osd-creds-mgmt-osd-staging-1-osdmanagedadminsre-secret -n aws-account-operator -o json | jq -r '.data.aws_access_key_id' | base64 -d); \
```
[the remark this answers is not in the document]
Comment (jharrington22) [reply Makefile]: @clcollins I'll do this in a separate PR as shell escaping could mean this breaks other test targets. Good suggestion.

**Comment by jharrington22** [comment]
@clcollins could you retest please?

**Comment by jharrington22** [comment]
> @jharrington22 - looks like it's not idempotent, and it leaves the account and account-related secrets.
> 
> Attempting to re-run the account reuse test fails because secrets associated with the accountclaim are missing from the new namespace. It looks like it fails get the output it was expecting at some point (note the empty secret name):
> 
> ```
> {"level":"error","ts":1586205522.4225044,"logger":"controller_accountclaim","msg":"Unable to find AWS account STS secret","Request.Namespace":"test-claim-namespace","Request.Name":"test-claim","error":"Secret \"\" not found",
> ```

@clcollins thank you for testing. I removed the target delete-account from the end of `test-reuse` so it wasn't deleting the account, its now idempotent.

**Comment by clcollins** [comment]
Beautiful, everything is working great @jharrington22 

/lgtm

**Pull request merged**